# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/baselsalah342-max/flyrank_intern/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

I am choosing Lane 2: Refresh / Content Opportunity Scoring. This lane is highly valuable because optimizing and refreshing existing content offers a faster and more efficient return on investment (ROI) for clients compared to creating entirely new content from scratch. It directly addresses editorial capacity constraints by prioritizing high-impact pages that need attention.

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
import pandas as pd


df = pd.read_csv("../../data/raw/content_refresh_anonymized.csv")


total_pages = df['content_id'].nunique()


declining_pages = df[df['trend_direction'] == 'down']['content_id'].nunique()
pct_declining = (declining_pages / total_pages) * 100


print(f"Total unique pages to analyze: {total_pages}")
print(f"Pages with a declining trend: {declining_pages} ({pct_declining:.2f}%)")

Total unique pages to analyze: 30000
Pages with a declining trend: 16262 (54.21%)


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

The Search Question: Which individual content pages are showing evidence of traffic decay or holding untapped visibility potential, and should be prioritized for a refresh review?

Unit of Analysis: A single content item per client (content_id / content_hash_id).

The Decision & Action: This work improves the editorial team's scheduling decision. Instead of guessing which pages to update, an editor will act directly on our ranked opportunity queue, reviewing the top-K pages alongside their specific reason codes to update, expand, or protect them.

The Cost of a Wrong Call: A false positive (recommending a healthy, stable page for a refresh) wastes limited human editorial hours on unnecessary rewrites. Even worse, it risks breaking the page's current successful structure, causing an accidental drop in its stable search engine ranking.

Why ML/Data helps: With thousands of content items across multiple clients, manually tracking continuous trends, keyword-mix decay, and engagement drops simultaneously is impossible for human teams. Machine learning and data pipelines can systematically calculate multi-signal risk profiles to rank these opportunities at scale.

In [2]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
pages_per_client = df.groupby('client_id')['content_id'].count()

print("Sample of content pieces a single client team has to manually monitor:")
print(pages_per_client.head())
print(f"\nAverage pages per client: {pages_per_client.mean():.1f}")

Sample of content pieces a single client team has to manually monitor:
client_id
client_02d20bbd7e      38
client_0b918943df      35
client_19581e27de    7008
client_1a6562590e       3
client_25fc0e7096     476
Name: content_id, dtype: int64

Average pages per client: 937.5


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [3]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
high_demand_decay = df[(df['impressions_90d'] >= 500) & (df['trend_direction'] == 'down')]['content_id'].nunique()


stale_top_pages = df[(df['avg_position'] <= 10) & (df['content_age_days'] >= 180)]['content_id'].nunique()


total_pages = df['content_id'].nunique()

print(f"Total Unique Pages evaluated: {total_pages}")
print(f"1. High-Demand Pages actively decaying: {high_demand_decay}")
print(f"2. Stale Page-One Content at high decay risk: {stale_top_pages}")

Total Unique Pages evaluated: 30000
1. High-Demand Pages actively decaying: 9961
2. Stale Page-One Content at high decay risk: 7810


## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

### Data Evidence for Lane 2

Based on the initial audit of the 30,000 unique pages in our starter dataset, we uncovered substantial empirical proof that validates the urgency of an automated refresh scoring model:

* **Massive Active Bleed (9,961 Pages):** Exactly 9,961 pages (33.2% of the entire catalog) have high market visibility with over 500 impressions, yet they are locked in an active downward traffic trend. These represent critical revenue leaks that our ranked queue will immediately flag for recovery.
* **High-Value Silent Decay Risk (7,810 Pages):** There are 7,810 pages currently holding premium organic real estate on Page 1 of search engines (average position <= 10) but haven't been updated in over 180 days. These are stable assets sitting in a danger zone, waiting to be overtaken by fresher competitor content.

Manually auditing 30,000 pages to catch these cross-sections is impossible for editorial teams. Building a system to rank and sort these ~17,700 urgent cases justifies the next 7 weeks of development.

What this work CAN claim (Observed & Decision Support):
This model provides a systematic ranking of existing content based on observable, structural signals (such as historical impression drops and content age).

It acts purely as a decision-support tool to surface high-probability opportunity pockets for human editors.

What this work CANNOT claim (Causal Proof & 'Predicting Google'):

This work cannot guarantee that updating a page will automatically restore its exact traffic or ranking; search algorithms involve dynamic external factors outside our dataset.

It establishes statistical correlations and historical risk trends, not definitive causal proof of why Google dropped a specific page.

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## Self-check

Before you submit, confirm each line honestly:

- [x] Every section above is filled — markdown thinking AND the code that backs it
- [x] The notebook runs top to bottom with no errors (Runtime → Run all)
- [x] No client names, URLs, or private queries anywhere
- [x] My claims use careful words: observed, measured, directional, decision-support
- [x] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.